onnx/backend/test/case/node/__init__.py

```diff
@@ -156,7 +156,16 @@ def expect(node,  # type: onnx.NodeProto
         atol=1e-7,
     ))
 
-    expanded_function_nodes = function_testcase_helper(node, name)
+    # Create list of types for node.input, filling a default TypeProto for missing inputs:
+    def merge(node_inputs, present_value_info):  # type: (List[Text], List[onnx.ValueInfoProto]) -> List[TypeProto]
```

Comment:
I dont understand what this is exactly trying to do. Can you please elaborate... 

For example when node_inputs does not exist why return [] and not [TypeProto()] 
Also if (node_inputs[0] != ''): what does this check really accomplish?

Reply:
I added comments to the beginning of the function to clarify what the inputs/outputs look like. This is needed because of the existing style of creating test-cases which omit values for optional-inputs when they are missing. So, we need to adjust for missing-optional-parameters when creating the list of input-types.